scalding-core/src/main/scala/com/twitter/scalding/serialization/MurmerHashUtils.scala

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import java.io.InputStream
+import scala.reflect.ClassTag
+import scala.collection.generic.CanBuildFrom
+import java.lang.Integer.{ rotateLeft => rotl }
+
+// Be careful using this, the product/array or similar will attempt to call system hash codes.
+import scala.util.hashing.MurmurHash3
+
+// Taking a few functions from:
+// https://guava-libraries.googlecode.com/git/guava/src/com/google/common/hash/Murmur3_32HashFunction.java
+object MurmerHashUtils {
+  final val seed = 0xf7ca7fd2
+  private final val C1: Int = 0xcc9e2d51
+  private final val C2: Int = 0x1b873593
+
+  final def hashInt(input: Int): Int = {
+    val k1 = mixK1(input)
+    val h1 = mixH1(seed, k1)
+
+    fmix(h1, 4) // length of int is 4 bytes
+  }
+
+  final def hashLong(input: Long): Int = {
+    val low = input.toInt
+    val high = (input >>> 32).toInt
+
+    var k1 = mixK1(low)
+    var h1 = mixH1(seed, k1)
+
+    k1 = mixK1(high)
+    h1 = mixH1(h1, k1)
+
+    fmix(h1, 8) // 8 bytes
+  }
+
+  final def hashUnencodedChars(input: CharSequence): Int = {
```

Comment:
This is straight out of guava, we aren't using it yet actually we use the String one in the scala murmur hash at the moment